pocs/utils/__init__.py

```diff
@@ -48,6 +48,42 @@ def flatten_time(t):
     return t.isot.replace('-', '').replace(':', '').split('.')[0]
 
 
+# This is a streamlined variant of PySerial's serialutil.Timeout.
+class Timeout(object):
+    """Simple timer object for tracking whether a time duration has elapsed.
+
+    Attribute `is_non_blocking` is true IFF the duration is zero.
+    """
+
+    def __init__(self, duration):
+        """Initialize a timeout with given duration (seconds)."""
+        assert duration >= 0
+        self.is_non_blocking = (duration == 0)
+        self.duration = duration
+        self.restart()
+
+    def expired(self):
+        """Return a boolean, telling if the timeout has expired."""
+        return self.time_left() <= 0
+
+    def time_left(self):
+        """Return how many seconds are left until the timeout expires."""
+        if self.is_non_blocking:
+            return 0
+        else:
+            delta = self.target_time - time.monotonic()
```

Comment:
Am I missing something about where these come from? Should this class inherit from `serialutil.Timeout`?

Reply:
I decided to just copy their code and eliminate things we don't need. time.monotonic() is like time.time(), but guarantees that it won't go backwards.

Reply:
Ok, I didn't realize `target_time` first gets set in the `restart` method so didn't see where it was coming from. Makes sense now.

Reply:
Although maybe just `start` is a better name?

Reply:
restart because __init__ calls it, so that it is started at creation. If too confusing, I could remove restart, so you have to create a new instance each time, which is definitely the common way to use it.